src/xenia/kernel/xam/xam_input.cc

```diff
@@ -118,6 +120,10 @@ dword_result_t XamInputSetState(dword_t user_index, dword_t unk,
     actual_user_index = 0;
   }
 
+  if (!cvars::enable_vibration) {
+    vibration->left_motor_speed = 0;
+    vibration->right_motor_speed = 0;
+  }
   auto input_system = kernel_state()->emulator()->input_system();
   return input_system->SetState(user_index, vibration);
```

Comment:
Ideally this should probably be handled by a proxy function in the HID subsystem, I think, also with the possibility of toggling it at runtime, zeroing the current speed when vibration is disabled mid-game from the UI, and remembering the speeds from the latest XamInputSetState call and applying it when vibration is re-enabled.

Reply:
I am unsure if it is necessary to have the rumble speed remembered. What guest do you now that has active rumble (at the same level) for more than a second? (while there is active user input)

Reply:
Environmental effects such as thunder possibly, big explosions in scripted scenes.

Reply:
Overall, statically, in the XInput API, there are no requirements of how frequently the function needs to be called so that we can treat a specific maximum length as "negligible". By only dropping calls, we're introducing arbitrary behavior not present in the API specification, as well as unspecified constraints on arbitrarily chosen dynamic state (when the game performs the XInputSetState calls, and the point in time when the user has toggled vibration). A properly isolated implementation should store the desired vibration level, and perform the toggling with the event flow similar to that of a new keystroke arriving from the UI thread in the `winkey` HID backend.